controller/controller.go

```diff
@@ -1404,10 +1404,14 @@ func (ctrl *ProvisionController) provisionClaimOperation(ctx context.Context, cl
 	if nodeName, ok := getString(claim.Annotations, annSelectedNode, annAlphaSelectedNode); ok {
 		if ctrl.nodeLister != nil {
 			selectedNode, err = ctrl.nodeLister.Get(nodeName)
+			// if node does not exist, remove volume.kubernetes.io/selected-node annotation
+			if apierrs.IsNotFound(err) {
```

Comment:
There is an interesting corner case. What if the provisioning did start already on the backend *and* the Node got deleted afterwards? Should we do non-final error handling instead and wait for the driver to return a final error?

cc @jsafrane

Reply:
I can imagine `provisionClaimOperation()` to check if the PVC is in `claimsInProgress` and keeps calling the CSI driver with an unavailable node until it gets a final answer. But it would need to store the `Node` _somewhere_.

I mean, it can be solved, I am not sure this is the right PR.

Reply:
So you mean we can ignore final/non-final handling for now, and look into it later?

Reply:
If the provisioning has started, shouldn't the state changed to ProvisioningInBackground?

If it is ProvisioningInBackground and node is not found, we do not go down the code path: return ctrl.provisionVolumeErrorHandling(ctx, ProvisioningReschedule, err, claim, operation)

Instead we continue to the code:
```
err = fmt.Errorf("failed to get target node: %v", err)
ctrl.eventRecorder.Event(claim, v1.EventTypeWarning, "ProvisioningFailed", err.Error())
return ProvisioningNoChange, err
```